src/Microsoft.OData.Client/DictionaryExtensions.cs

```diff
@@ -54,6 +55,23 @@ internal static class DictionaryExtensions
             }
         }
 
+        /// <summary>
+        /// Convenience funcion that wraps ConcurrentDictionary.TryAdd() to allow same signature as IDictionary
+        /// </summary>
+        public static void Add<TKey, TValue>(this ConcurrentDictionary<TKey, TValue> self, TKey key, TValue value)
+        {
+            if(!self.TryAdd(key, value))
+                throw new ArgumentException("Argument_AddingDuplicate");
+        }
+
+        /// <summary>
+        /// Convenience function for ConcurrentDictionary to allow same signature as IDictionary

```

Comment:
I am wondering why we need this wrapper, since IDictinoary<TKey, TValue>.Remove(TKey) already exists?

Reply:
Because Remove is an explicitly implemented member of IDictionary<TKey,TValue>; you have to cast to IDictionary<TKey,TValue> to call it.  This helper method avoids having to cast where Remove is called.

---
In reply to: [166143263](https://github.com/OData/odata.net/pull/1054#discussion_r166143263) [](ancestors = 166143263)